contracts/modules/maker/MakerV2Invest.sol

```diff
@@ -0,0 +1,178 @@
+pragma solidity ^0.5.4;
+
+import "./MakerV2Base.sol";
+import "../../defi/Invest.sol";
+
+contract PotLike {
+    function chi() public view returns (uint);
+    function pie(address) public view returns (uint);
+    function drip() public;
+}
+
+/**
+ * @title MakerV2Invest
+ * @dev Module to lock/unlock MCD DAI into/from Maker's Pot
+ * @author Olivier VDB - <olivier@argent.xyz>
+ */
+contract MakerV2Invest is Invest, MakerV2Base {
+
+    // The address of the Pot
+    PotLike internal pot;
+
+    bytes4 constant internal ADAPTER_JOIN = bytes4(keccak256("join(address,uint256)"));
+    bytes4 constant internal ADAPTER_EXIT = bytes4(keccak256("exit(address,uint256)"));
+    bytes4 constant internal VAT_HOPE = bytes4(keccak256("hope(address)"));
+    bytes4 constant internal POT_JOIN = bytes4(keccak256("join(uint256)"));
+    bytes4 constant internal POT_EXIT = bytes4(keccak256("exit(uint256)"));
+
+    // *************** Constructor ********************** //
+
+    constructor(
+        PotLike _pot
+    )
+        public
+    {
+        pot = _pot;
+    }
+
+    // *************** External/Public Functions ********************* //
+
+    /* ********************************** Implementation of Invest ************************************* */
+
+    /**
+     * @dev Invest tokens for a given period.
+     * @param _wallet The target wallet.
+     * @param _token The token address.
+     * @param _amount The amount of tokens to invest.
+     * @param _period The period over which the tokens may be locked in the investment (optional).
+     * @return The exact amount of tokens that have been invested.
+     */
+    function addInvestment(
+        BaseWallet _wallet,
+        address _token,
+        uint256 _amount,
+        uint256 _period
+    )
+        external
+        onlyWalletOwner(_wallet)
+        onlyWhenUnlocked(_wallet)
+        returns (uint256 _invested)
+    {
+        require(_token == address(daiToken), "DM: token should be DAI");
+        joinDsr(_wallet, _amount);
+        _invested = _amount;
+        emit InvestmentAdded(address(_wallet), address(daiToken), _amount, _period);
+    }
+
+    /**
+     * @dev Exit invested postions.
+     * @param _wallet The target wallet.
+     * @param _token The token address.
+     * @param _fraction The fraction of invested tokens to exit in per 10000.
+     */
+    function removeInvestment(
+        BaseWallet _wallet,
+        address _token,
+        uint256 _fraction
```

Comment:
Are we sure we want to stick with `fraction` instead of `amount`? Given our current strategy there is no need for a generic interface. We may want to check with the clients if it would solve problems or if they are fine as it is.

Reply:
Being discussed with @DarthMike and @sgc-code. Note that if we use `_amount`, we'll have to also add a `removeAllInvestment()` external method.

Reply:
Happy to have `removeInvestment` use exact `amount` (and event to return an amount as well). Currently users have issue with Compound and DSR. 

My concern was not changing Compound API as it should be the same as well, and we would fix all rounding issues we've seen.